### PR TITLE
Embedded domain field into unstructured 2D mesh struct.

### PR DESCRIPTION
**Pull request by sriharshakandala**

### CODE REVIEW

**Comment by jakebolewski** [comment]
@sriharshakandala I think this needs to be rebased after https://github.com/CliMA/ClimaCore.jl/pull/201 is merged.  Looking at it though I don't think it will be too bad, also #201 allows for the coordinate types defined in #81 to be applied for the sphere.

File: src/Meshes/Meshes.jl [hunk 1]
```diff
@@ -184,7 +183,9 @@ https://p4est.github.io/papers/BursteddeWilcoxGhattas11.pdf
 # Fields
 $(DocStringExtensions.FIELDS)
 """
-struct Mesh2D{I, IA1D, IA2D, FT, FTA2D, SNT, NB} <: AbstractMesh{FT}
+struct Mesh2D{I, IA1D, IA2D, FT, FTA2D, SNT, NB, D} <: AbstractMesh{FT}
```
Comment (jakebolewski) [inline src/Meshes/Meshes.jl:186]: usually the unique types are in the order of the fields to make it easy to reference

File: src/Domains/Domains.jl [hunk 2]
```diff
@@ -125,14 +125,22 @@ end
 coordinate_type(::RectangleDomain{FT}) where {FT} =
     Geometry.Cartesian2DPoint{FT}
 
+abstract type AbstractSphere{FT} end
+struct EquiangularSphere{FT} <: AbstractSphere{FT} end
```
Comment (jakebolewski) [inline src/Domains/Domains.jl:129]: isn't the sphere type more a property of the mesh than the domain?

Comment (jakebolewski) [reply src/Domains/Domains.jl:129]: ex. an `EquiangularSphereMesh()` makes sense but a equiangular sphere domain does not really as the domain just defines the coordinates (LatLong) / coordinate extent (Radius).

**Comment by sriharshakandala** [comment]
@jakebolewski : I rebased the code with master. I also added the bickley jet CG experiment with unstructured mesh to the buildkite so that changes relevant to unstructured mesh will also be carried over wherever newer PRs are added.

**Comment by sriharshakandala** [comment]
bors try

**Comment by jakebolewski** [comment]
Nice, seems to be a bit simpiler and is more in line with what we do for the vertical warping. LGTM

**Review by jakebolewski — APPROVED** [review]

**Comment by sriharshakandala** [comment]
Bots r+

**Pull request merged**